src/ModelContextProtocol/Client/McpClientTool.cs

```diff
@@ -9,21 +9,50 @@ namespace ModelContextProtocol.Client;
 public sealed class McpClientTool : AIFunction
 {
     private readonly IMcpClient _client;
+    private readonly string? _nameOverride;
+    private readonly string? _descriptionOverride;
 
-    internal McpClientTool(IMcpClient client, Tool tool)
+    internal McpClientTool(IMcpClient client, Tool tool, string? nameOverride = null, string? descriptionOverride = null)
     {
         _client = client;
         ProtocolTool = tool;
+        _nameOverride = nameOverride;
+        _descriptionOverride = descriptionOverride;
```

Comment:
If we're adding the fields anyway, wouldn't it be simpler to do it like so?

```suggestion
        _name = nameOverride ?? tool.Name ?? string.Empty;
        _description = descriptionOverride ?? ProtocolTool.Description ?? string.Empty;
```

And then just use `=> _name;` and `=> _description;` as the getters?

I'd also want to consider what's so bad about just making these properties publicly settable. I know it's not exactly pure, but we don't have to make any promises about what happens if you set it. Or we could promise to ignore whatever you set which is essentially what we're doing with this.

Reply:
I wouldn't mind public setters personally, but I had the impression MEAI had immutable AIFunctions and thus wanted to align with that.

Reply:
Making these mutable seems reasonable.

Reply:
AIFunction has no WithName or WithDescription members to override either. You need to have a McpClientTool to use it. Isn't it the same with a normal setter?

Reply:
I don't think so :(

Reply:
> AIFunction has no WithName or WithDescription members to override either. You need to have a McpClientTool to use it. Isn't it the same with a normal setter?

But that's ok - this is for MCP host developers who need to adjust details of McpClientTools before passing them on as AIFunctions. I think making this a slightly awkward operation is ok - it should be used for a reason. The old C++ `reinterpret_cast<T>` pattern of ensuring someone who uses this feature actually needs it, without making it too annoying.

Because while I seems like an exotic use case - when one does hit it, it's crippling not to be able to override these properties - and having to write a carbon copy of McpClientTool and the handler extensions isn't great ergonomics.

However, I am certainly open to something that ends up looking more idiomatic (and I actually think being to override a virtual getter while also providing a new setter should be supported by the language - because the alternative is gunky stuff like this - but I guess there are permutations where the setter is hidden because of access modifiers that make this problematic).